services/elasticache/pkg/resource/cache_parameter_group/custom_api.go

```diff
@@ -0,0 +1,225 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cache_parameter_group
+
+import (
+	"context"
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	ackerr "github.com/aws/aws-controllers-k8s/pkg/errors"
+	svcapitypes "github.com/aws/aws-controllers-k8s/services/elasticache/apis/v1alpha1"
+	svcsdk "github.com/aws/aws-sdk-go/service/elasticache"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// describeCacheParameters returns Cache Parameters for given Cache Parameter Group name and source
+func (rm *resourceManager) describeCacheParameters(
+	ctx context.Context,
+	cacheParameterGroupName *string,
+	source *string,
+) ([]*svcapitypes.ParameterNameValue, error) {
+	parameterNameValues := []*svcapitypes.ParameterNameValue{}
+	var paginationMarker *string = nil
+	for {
+		input, err := rm.newDescribeCacheParametersRequestPayload(cacheParameterGroupName, source, paginationMarker)
+		if err != nil {
+			return nil, err
+		}
+		response, respErr := rm.sdkapi.DescribeCacheParametersWithContext(ctx, input)
+		rm.metrics.RecordAPICall("READ_ONE", "DescribeCacheParameters", respErr)
```

Comment:
technically this is a READ_MANY call, not a READ_ONE. The resource being read is the cache parameters, not the cache parameter group.

Reply:
Updated the code.